datalad_deprecated/resources/website/assets/js/main.js

```diff
@@ -530,6 +530,40 @@ function directory(jQuery, md5) {
       }
       jQuery('td', row).eq(0).html(orig);
 
+      var menu = '<div class="context-menu">' +
+                  '<button class="context-button">&#10247;</button>' +
+                  '<ul class="context-content">'+
+                    '<li class="context-option copy">Copy Link</li>';
+      
+      // add rendering option
+      // https://stackoverflow.com/questions/14780350/convert-relative-path-to-absolute-using-javascript
+      var link = document.createElement("a");
+      link.href = traverse.next;
+      link.protocol = "https";
+
+      if (/.nii(.gz)?$/.test(traverse.next)) {
+        menu +=  '<li class="context-option render"><a href="https://bioimagesuiteweb.github.io/webapp/viewer.html?image=' + link.href + '">View in Renderer</a></li>';
+      }
+      else if (/.nwb$/.test(traverse.next)) {
+        menu +=  '<li class="context-option render"><a href="http://nwbexplorer.opensourcebrain.org/nwbfile=' + link.href + '">View in Renderer</a></li>';
+      }
```

Comment:
Let's not postpone properly separating the "spec"  of "renders" from code generating the menu. (and again -- code duplication is evil! ;))

- take that https://github.com/dandi/dandiarchive/wiki/WiP:-dump-of-thoughts-on-%22services-registry%22 and make it into a data structure on top of the script, e.g. "EXTERNAL_SERVICES" (just exclude/comment out "NWB Widgets (Binder):" for now).  Could be simplified or just harmonized (always a list) if desired to avoid that single item vs list, up to you.  Can do simple string replacement of `{url}` with actual url for now, since we have no other values to pass and you didn't want to use string interpolations 
- have a function given an item (`traverse.next` here?) return a dict of "name: URL" which you render here in the loop as `<li` after "External Services" section of a kind (just e.g. a bolded text in the menu, doesn't need to do anything upon click). Note -- name should be `Bioimagesuite/Viewer` etc, not just "View in Renderer" since a single type of file could have various.

Then later we might even have such data structure defined/fetched externally and code here would be ready to handle any new ones possibly without even redeploying anything.

Reply:
also -- make `traverse.next` called ones and assigned near `var traverse` since I  guess (didn't check) it would run server queries then for each invocation, which could be avoided.

Reply:
I added a data structure that should do that (though I still need to test more and check about some of the other file types, like directory) -- should be up on datasets-dev. I'll add the function you mentioned in point 2 soon.

As for the `traverse.next`, I think accessing that isn't a server query, only the initial `var traverse = clickhandler(data)` is.